inngest/examples/axum/main.rs

```diff
@@ -147,22 +174,38 @@ fn step_run() -> ServableFn<TestData, InngestError> {
             event: "test/step-run".to_string(),
             expression: None,
         },
-        |_input: &Input<TestData>,
-         step: &mut StepTool|
-         -> Result<serde_json::Value, InngestError> {
+        call_some_step_function,
+    )
+}
+
+fn incorrectly_propagates_error() -> ServableFn<TestData, InngestError> {
+    create_function(
+        FunctionOps {
+            id: "Step run".to_string(),
+            ..Default::default()
+        },
+        Trigger::EventTrigger {
+            event: "test/step-run-incorrect".to_string(),
+            expression: None,
+        },
+        |_input: Input<TestData>, step: StepTool| async move {
             let some_captured_variable = "captured".to_string();
 
-            let step_res = step.run(
-                "some-step-function",
-                || -> Result<serde_json::Value, UserLandError> {
+            let res = step
+                .run("some-step-function", || async move {
```

Comment:
ah memories are coming back to me now.
I really don't like these syntax of `async move` but it's not like there's much of a choice. lol

Reply:
lol, yeah, async closures are coming soon ™️ 
https://blog.rust-lang.org/inside-rust/2024/08/09/async-closures-call-for-testing.html

Reply:
PLEASE!!!

Reply:
well, but the life of the library is you need to support a large range of versions that are not EOF so it'll be a long time before that syntax can be used even when they do get it out :cry: